### ISSUE TITLE
Support automatic length inference

### ISSUE THREAD

**Issue by ajusa** [sealmove/binarylang#8]
From talking with sealmove on Discord. It would be nice to have a way to automatically bind/set an integer field based on the length of other fields. Many protocols send length and data together, so that the reader knows how many bytes to read before stopping/moving to the next field.
For example with the PascalString example, I should only need to pass in the actual string itself, and have some way in the DSL to show that I want `len` to be the same length as the `str` I passed in.

Ideally this would also be able to work with more complex examples.

**Comment by sealmove** [comment]
This could be implemented as a property (property system was just added).

```nim
createParser(PascalString):
  u8: len
  s {@bind: len = e.len}: str(len)
```

This will force the name of the second field to be generated with `genSym`. A getter/setter with the name `str` will be generated. The getter will just return the field, while the setter will set both the `str` field to the passed value, and `len` field to `obj.str.len`.

**Comment by sealmove** [comment]
Implemented with v0.2.0.
